Required `str` parsed as `'None'` when argument value is `""`

Test implemented in #24 

Fails with `AssertionError: assert 'None' == ''`:

```py
uv run --extra dev -- pytest tests/test_base.py::test_empty_required_str
================================================================================================================================================================================= test session starts ==================================================================================================================================================================================
platform linux -- Python 3.11.10, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/mick/projects/draccus
configfile: pyproject.toml
collected 1 item                                                                                                                                                                                                                                                                                                                                                                       

tests/test_base.py F                                                                                                                                                                                                                                                                                                                                                             [100%]

======================================================================================================================================================================================= FAILURES =======================================================================================================================================================================================
_______________________________________________________________________________________________________________________________________________________________________________ test_empty_required_str ________________________________________________________________________________________________________________________________________________________________________________

    def test_empty_required_str():
        @dataclass
        class Args:
            a: str
    
        cfg = draccus.parse(Args, args=["--a="])
>       assert cfg.a == ""
E       AssertionError: assert 'None' == ''
E         
E         + None

tests/test_base.py:159: AssertionError
=============================================================================================================================================================================== short test summary info ================================================================================================================================================================================
FAILED tests/test_base.py::test_empty_required_str - AssertionError: assert 'None' == ''
================================================================================================================================================================================== 1 failed in 0.04s ===================================================================================================================================================================================
```

--- a/draccus/parsers/config_parsers.py
+++ b/draccus/parsers/config_parsers.py
@@ -33,6 +33,10 @@ class YAMLParser(Parser):
 
         from .yaml_loader import SafeLoaderWithInclusion
 
+        # yaml interprets empty strings as None, so we need to check for that
+        if s == "":
+            return s
+
         return yaml.load(s, SafeLoaderWithInclusion)
 
     @staticmethod

--- a/tests/test_base.py
+++ b/tests/test_base.py
@@ -195,3 +195,13 @@ def test_argparse_with_custom_parsing():
     draccus.decode.register(SomeClass, lambda x: SomeClass(x[7:]))
 
     Config.setup("--s custom_bob")
+
+
+def test_empty_required_str():
+    @dataclass
+    class Args:
+        a: str
+
+    cfg = draccus.parse(Args, args=["--a="])
+
+    assert cfg.a == ""

